core/src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -407,7 +412,7 @@ void validateIncomingState(ClusterState incomingState, ClusterState lastSeenClus
             final String message = String.format(
```

Comment:
This check rejects incoming states from the same master that are out of order. I do not think that the new check covers this case. I think it's needed?